UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -846,28 +940,46 @@ protected virtual IEnumerator ConstructMesh(GLTF.Schema.Mesh mesh, Transform par
 				yield return ConstructMeshPrimitive(primitive, meshId, i, materialIndex);
 
 				var primitiveObj = new GameObject("Primitive");
-				var meshFilter = primitiveObj.AddComponent<MeshFilter>();
-				meshFilter.sharedMesh = _assetCache.MeshCache[meshId][i].LoadedMesh;
-				var meshRenderer = primitiveObj.AddComponent<MeshRenderer>();
 
 				MaterialCacheData materialCacheData =
 					materialIndex >= 0 ? _assetCache.MaterialCache[materialIndex] : _defaultLoadedMaterial;
 
-				UnityEngine.Material material = materialCacheData.GetContents(primitive.Attributes.ContainsKey(SemanticProperties.Color(0)));
-
-				meshRenderer.material = material;
+				UnityEngine.Material material = materialCacheData.GetContents(primitive.Attributes.ContainsKey(SemanticProperties.Color(0)));
+
+				UnityEngine.Mesh curMesh = _assetCache.MeshCache[meshId][i].LoadedMesh;
+				if (NeedsSkinnedMeshRenderer(primitive, skin))
+				{
+					var skinnedMeshRenderer = primitiveObj.AddComponent<SkinnedMeshRenderer>();
+					skinnedMeshRenderer.material = material;
+					skinnedMeshRenderer.quality = SkinQuality.Bone4;
```

Comment:
should this be a modifiable property on the script?

Reply:
Do you mean that adding the skinnedmeshrenderer should be modifiable? I think it's correct to only add it if the mesh has a skin.  We don't have switches for other things like whether we add a material or not, so I think this is a similar situation

I did realize after you pointed out this code that the mesh filter is only added to the static objects, so I've fixed that :)

Reply:
No meant whether the skinquality should be